Doc/library/re.rst

```diff
@@ -710,6 +713,9 @@ form.
    string and immediately before the newline (if any) at the end of the string.
    Corresponds to the inline flag ``(?m)``.
 
+.. data:: NOFLAG
+
+   Indicates no flag being applied, the value is ``0``.
```

Comment:
Similarly, explain when this is useful and add `versionadded`.

Reply:
It can be used for example as a default value for a function parameter, and also as a base value that will later conditionally be ORed with other flags. But this explanation would be confusing to many readers without code examples.

An example as follows may be enough:

def myfunc(text, flag=re.NOFLAG):
    return re.match(text, flag)

What do you think?

Reply:
Thanks! I think that's a useful example.

Reply:
When was it added?

It seems that the discussion in #57594 was ignored, plus the naming is slightly stange (_no flags_ reads better than _no flag_)

Reply:
Not ignored, missed.  It took place 11 years prior.  We can certainly add a `NOFLAGS` name now, and make `NOFLAG` be an alias.